### ISSUE TITLE
5 minute delay in storage_picker.rb

### ISSUE THREAD

**Issue by generalinterest** [cloudfoundry/bosh-vsphere-cpi-release#163]
## Support Question
Environment: vSphere 6.0.0 with three clusters in a single Datacenter, and a single datastore in each cluster.

Problem: A Bosh Director with vSphere CPI v50 experiences a 5 minute delay when bosh recreating a VM.  A Bosh Director with the CPI v45.1.0 does not have this delay.
 
From the task debug log  following the "Starting create_vm..."

I, [2018-10-13T12:19:16.943111 #13837]  INFO -- [req_id cpi-475005]: Using global ephemeral disk datastore pattern: ^(MYDATASTORE)$
D, [2018-10-13T12:24:31.076307 #13837] DEBUG -- [req_id cpi-475005]: Initial Storage Options for creating ephemeral disk from pattern: ["MYDATASTORE"]
D, [2018-10-13T12:24:31.076500 #13837] DEBUG -- [req_id cpi-475005]: Storage Options for creating ephemeral disk are: ["MYDATASTORE"]
I, [2018-10-13T12:24:31.076621 #13837]  INFO -- [req_id cpi-475005]: Checking if ip '10.1.2.3' is in use

Is there anyway to account whats happening during this delay, to look and avoid it?

Regards

Brian.

---

You can also reach out to us via:

* Slack: http://cloudfoundry.slack.com/messages/bosh-cpi-dev

**Comment by EleanorRigby** [comment]
Can you provide complete CPI log for v45 and v50 , where you see the delay?

Are BOSH version , stemcells version different or same while using these
different CPI releases?

**Comment by generalinterest** [comment]
Hi

Attaching the CPI logs from a bosh recreate job run on three foundations.

There is a bump in the BOSH versions and Stemcells...

The CPI v45 is BOSH version 265.11.0, stemcell 3541.49

The CPI v50 is BOSH version 266.12, stemcell 3586.43 

I have the anonymised logs for one cpi v45 foundation, and two for v50 foundations.

2018-10-22_pcf_2.1.16_recreate_task_17563_cpi  is the v45 foundation

2018-10-22_pcfd_2.2.7_recreate_task_36256_cpi
2018-10-22_pcfsb_2.2.7_recreate_task_2349376_cpi   are the two v50 foundations.

The original Issue came from the debug log for the task 2349376.


[2018-10-22_pcf_2.1.16_recreate_task_17563_cpi.txt](https://github.com/cloudfoundry/bosh-vsphere-cpi-release/files/2501708/2018-10-22_pcf_2.1.16_recreate_task_17563_cpi.txt)
[2018-10-22_pcfd_2.2.7_recreate_task_36256_cpi.txt](https://github.com/cloudfoundry/bosh-vsphere-cpi-release/files/2501709/2018-10-22_pcfd_2.2.7_recreate_task_36256_cpi.txt)
[2018-10-22_pcfsb_2.2.7_recreate_task_2349376_cpi.txt](https://github.com/cloudfoundry/bosh-vsphere-cpi-release/files/2501710/2018-10-22_pcfsb_2.2.7_recreate_task_2349376_cpi.txt)




--

**Comment by nehagjain15** [comment]
Thank you for the logs, we will take a look at it and get back to you. If it is possible can you please try the latest version of CPI v51 ? We did a re-architecture for VM/Disk Placement as part of that release and it has lot more debugging log to help find issues with placement.

**Comment by yeshwantbabar** [comment]
@nehagjain15 @EleanorRigby  - Is there any update on this?

**Comment by nehagjain15** [comment]
@generalinterest I tested with v50 on 6.0 env with 2 clusters and each has multiple datastores and did not see any delay. You might want to look into your VC logs to see what call took long time.
```I, [2018-10-29T21:39:23.989779 #495]  INFO -- [req_id cpi-879705]: Starting create_vm...
D, [2018-10-29T21:39:23.991406 #495] DEBUG -- [req_id cpi-879705]: Running method 'RetrieveServiceContent'...
D, [2018-10-29T21:39:24.024429 #495] DEBUG -- [req_id cpi-879705]: Running method 'Login'...
D, [2018-10-29T21:39:24.079489 #495] DEBUG -- [req_id cpi-879705]: Running method 'FindByInventoryPath'...
D, [2018-10-29T21:39:24.083737 #495] DEBUG -- [req_id cpi-879705]: Running method 'CreateContainerView'...
D, [2018-10-29T21:39:24.100045 #495] DEBUG -- [req_id cpi-879705]: All clusters provided: {"vcpi-cluster-1"=>#<VSphereCloud::ClusterConfig:0x000055c2ba7231c0 @name="vcpi-cluster-1", @config={}>}
D, [2018-10-29T21:39:24.100126 #495] DEBUG -- [req_id cpi-879705]: Running method 'FindByInventoryPath'...
I, [2018-10-29T21:39:24.123438 #495]  INFO -- [req_id cpi-879705]: Using global ephemeral disk datastore pattern: nfs0-1
D, [2018-10-29T21:39:26.102467 #495] DEBUG -- [req_id cpi-879705]: Initial Storage Options for creating ephemeral disk from pattern: ["nfs0-1"]
D, [2018-10-29T21:39:26.102599 #495] DEBUG -- [req_id cpi-879705]: Storage Options for creating ephemeral disk are: ["nfs0-1"]
I, [2018-10-29T21:39:26.102683 #495]  INFO -- [req_id cpi-879705]: Checking if ip '192.168.111.158' is in use
D, [2018-10-29T21:39:26.102746 #495] DEBUG -- [req_id cpi-879705]: Running method 'FindByIp'...
I, [2018-10-29T21:39:26.106644 #495]  INFO -- [req_id cpi-879705]: No IP conflicts detected
I, [2018-10-29T21:39:26.106802 #495]  INFO -- [req_id cpi-879705]: Creating vm: vm-6dc6a864-0b7d-4487-8226-19670db0fbd8 on <[Vim.ClusterComputeResource] domain-c14> stored in <[Vim.Datastore] datastore-58> ```

**Comment by nehagjain15** [comment]
@generalinterest  this has been fixed in v50.0.3. Please try that.

**Comment by EleanorRigby** [comment]
Feel free to open this if v50.0.3 does not address your concern.
